### DISCUSSION TITLE
P4A: Make Android permissions optional/configurable

### DISCUSSION THREAD

**Discussion by vabt-igd** [hbldh/bleak#1337] [Ideas]
Hey,

would you be willing to make the permissions in the Android backend optional or configurable? 
As can be read [here](https://developer.android.com/guide/topics/connectivity/bluetooth/permissions), it is not always necessary nor desired to ask for [Permission.ACCESS_FINE_LOCATION, Permission.ACCESS_COARSE_LOCATION, "android.permission.ACCESS_BACKGROUND_LOCATION"]. 
On newer devices and depending on the use case, it may be sufficient to ask for [(name=android.permission.BLUETOOTH_SCAN;usesPermissionFlags=neverForLocation), android.permission.BLUETOOTH_CONNECT]. 

If you were willing to accept such a change, I would be happy to contribute with a PR and do some minor testing (I currently only have access to two android devices running on API level 31 and 32).

**Comment by dlech** [comment]
If you can make it work for both old and new Android API levels at the same time, happy to look at a PR.

**Comment by vabt-igd** [comment]
Perfect. I'll see, if I can come up with a solution and will ping you in a PR, when ready.
